File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int N = Integer.parseInt(br.readLine());

        int count = 1;

        ArrayList<String> answer = new ArrayList<>();

        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < N; i++) {
            int a = Integer.parseInt(br.readLine());
            while (a >= count) {
                stack.add(count);
                answer.add("+");

                count++;
            }

            if (a == stack.lastElement()) {
                stack.pop();
                answer.add("-");

            } else {
                answer.clear();
                answer.add("NO");
                break;
            }

        }
        
        answer.forEach(i -> sb.append(i).append("\n"));
        System.out.println(sb);
        
    }


}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

input = sys.stdin.readline

N = int(input())

nums = [x for x in range(1, N + 1)]

stack = []

answer = []

count = 1

for _ in range(N):  
    a = int(input()) 
    while a >= count:  
        stack.append(count)  
        answer.append("+")
        count += 1
    if stack[-1] == a:
        answer.append("-")
        stack.pop()
    else:
        answer = ["NO"]
        break

list(map(lambda x: print(x), answer))
